Automattic/wp-calypso
# [PBD interim fix] Add support for SITE_UNLAUNCHED to hold list

#### Changes proposed in this Pull Request

Currently an unlaunched site returns the `SITE_PRIVATE` response from the AT eligibility API. This is technically true, an unlaunched site is a kind of private. However after D36437-code ships the API will make a distinction between unlaunched and private (this is to facilitate the work in #38141).

In the time between D36437-code and #38141 shipping the `<EligibilityWarnings>` component should continue to look the same.

* Deserialize `SITE_UNLAUNCHED` response from eligibility API
* Show identical messages for `SITE_PRIVATE` and `SITE_UNLAUNCHED`

#### Testing instructions



* Apply D36437-code
* Create new simple site
* Attempt to upload a plugin
* See the "Public site needed" message
* Launch your site and make it private
* Try and upload plugin again
* See the exact same "Public site needed" message
* Make site public
* Try and upload plugin again
* The "Public site needed" message has disappeared
* Do the same but try to upload a theme
* Do the same but give the site a business plan and try to adjust hosting options (this is a new way to initiate an AT)
